src/applications/static-pages/events/helpers/index.js

```diff
@@ -64,20 +64,70 @@ export const deriveMostRecentDate = (
   return futureDates[0];
 };
 
-export const removeDuplicateEvents = events => {
-  return events?.reduce((allEvents, event) => {
-    const currentEventIds = allEvents.map(
-      currentEvent => currentEvent.entityId,
-    );
+function hideEventRepeatsFromList(allEvents, event) {
+  if (!allEvents) {
+    return [];
+  }
+
+  if (allEvents === []) {
+    return [event];
+  }
+
+  const currentEventIds = allEvents.map(ev => ev.entityId);
+
+  if (!currentEventIds.includes(event.entityId)) {
+    allEvents.push(event);
+  }
+
+  return allEvents;
+}
 
-    return !currentEventIds.includes(event.entityId)
-      ? [...allEvents, event]
-      : allEvents;
-  }, events);
+export const removeDuplicateEvents = events =>
+  events?.reduce(hideEventRepeatsFromList, []);
+
+// This takes all repeating events and creates a separate event for
+// each repeated instance. Repeating events can still be identified as such,
+// and let event listings show multiple recurring events.
+export const fleshOutRecurringEvents = events => {
+  if (!events) {
+    return [];
+  }
+
+  const allEvents = events.reduce((fullEvents, event) => {
+    if (!event.fieldDatetimeRangeTimezone) {
+      return fullEvents;
+    }
+
+    if (event?.fieldDatetimeRangeTimezone.length === 1) {
+      fullEvents.push(event);
+      return fullEvents;
+    }
+
+    const eventTimes = event?.fieldDatetimeRangeTimezone;
+    // This makes each copy of a recurring event start with a different time,
```

Comment:
It doesn't need to retain the entire list, but the array needs to have more than one item since array length is how the code knows an event is recurring. Keeping the entire array after re-ordering makes more sense since it at least makes things easier to debug by looking at the data.

Reply:
Understood!